services/QuillLMS/app/services/google_integration/user.rb

```diff
@@ -28,25 +23,29 @@ def update_or_initialize
       params = {
         signed_up_with_google:      true,
         post_google_classroom_assignments: user.new_record? ? true : user.post_google_classroom_assignments,
-        auth_credential_attributes: auth_credential_attributes,
+        auth_credential_attributes: auth_credential_attributes(user),
       }
 
-      params[:name]      = data.name      if data.name.present?
-      params[:email]     = data.email     if data.email.present?
-      params[:google_id] = data.google_id if data.google_id.present?
-      params[:role]      = data.role      if data.role.present? && user.new_record?
+      params[:name]      = profile.name      if profile.name.present?
+      params[:email]     = profile.email     if profile.email.present?
+      params[:google_id] = profile.google_id if profile.google_id.present?
+      params[:role]      = profile.role      if profile.role.present? && user.new_record?
       params[:clever_id] = nil
 
       params
     end
   end
 
-  private def auth_credential_attributes
+  private def find_user_by_google_id_or_email
```

Comment:
Optional: you could make this a class method that accepts params, then reuse it in `google_controller#verify_credentials`

Reply:
I actually toyed with extraction this morning, but opted not to...for now.